### PR TITLE
TS-3683: Add log tag for SSL Session/Ticket HIT and TCP connection reused

### PR DESCRIPTION
**Pull request by fpesce**
Additionally, a TSAPI function for those who used an external/plugin implementation for SSL Ticket/Session is provided.

### CODE REVIEW

File: doc/admin/event-logging-formats.en.rst [hunk 1]
```diff
@@ -170,6 +170,24 @@ The following list describes Traffic Server custom logging fields.
     mm is the two-digit minutes value, and ss is the 2-digit seconds
     value (for example, 16:01:19).
 
+.. _cqtr:
+
+``cqtr``
+    The TCP reused status; indicates if this request went through an
+    already established connection.
```
Comment (SolidWallOfCode) [inline doc/admin/event-logging-formats.en.rst]: Presume this is the client request? That should be explicitly mentioned.

File: doc/reference/api/TSNetVConnSSLCachedHitSet.en.rst [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+`#include <ts/ts.h>`
+
+.. c:function:: void TSNetVConnSSLCachedHitSet(TSVConn vc, int state)
+
+Description
+-----------
+
+   Mark an SSL Connection as reused, which means that an SSL Session Cache or an SSL Ticket have been used to prevent a full handshake for this SSL Connection.
+
```
Comment (SolidWallOfCode) [inline doc/reference/api/TSNetVConnSSLCachedHitSet.en.rst]: This API makes no sense to me. The core should know this and set the state accordingly. The better API would be to enable the plugin to control ticket use for an inbound connection but core should handle the ticket and tracking whether it was used. This is the same reason the cqtr log tag makes sense without any API to set that state.

**Comment by SolidWallOfCode** [comment]
Overall I think the log tags are useful in ways aggregate statistics are not and I support adding those tags.

**Comment by shinrich** [comment]
My only comment is about the limits of the client_* fields.  As they stand, they only accurately track HTTP/1.x over TCP/SSL, not requests over SPDY/H2.  I know that this is something additional you are working on and can be added in via an addition jira.

This pull request also needs to be rebased before it can be committed.

**Comment by shinrich** [comment]
Sorry I had a git log mishap pushing these changes.  I missed the "This closes" comment and the changes showed up in the following for comments.

{code}
da04362227ef91b27aa7d02e9238f1ceae68689d
f3e13664ab20f60cb4bd2ffef1eb7d6a374a1698
5a4350e6067ac868e54538467ec83a9413853143
71752c741ac8b49d432dd4b13f5ea2a7f176b37e
{code}

@fpesce could you go ahead and close this pull request?

**Comment by fpesce** [comment]
Done @shinrich :smiley:

**Pull request closed without merging**